examples/shop-page/Components/ShopPage.lua

```diff
@@ -0,0 +1,77 @@
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local Players = game:GetService("Players")
+
+local Components = ReplicatedStorage:WaitForChild("Components")
+
+local Roact = require(ReplicatedStorage.Roact)
+
+local ProductItemList = require(Components:WaitForChild("ProductItemList"))
+
+local excludePropsForFrame = {
+	items = true,
+	itemsPerRow = true,
+	itemAspectRatio = true,
+	padding = true
+}
+
+local ShopPage = Roact.Component:extend("ShopPage")
+
+function ShopPage:init()
+	self:setState({
+		cellSize = 100,
+		canvasHeight = 100,
+	})
+
+	self.onAbsoluteSizeChanged = function(frame)
+		local props = self.props
+		local state = self.state
+		local padding = props.padding or 0
+		local itemsPerRow = props.itemsPerRow
+
+		local totalWidth = frame.AbsoluteSize.X
+		local cellWidth = (totalWidth - padding * (itemsPerRow + 1)) / itemsPerRow
+		local cellHeight = cellWidth * props.itemAspectRatio
+		local rows = math.ceil(#props.items / itemsPerRow)
+		local canvasHeight = rows * cellHeight + padding * (rows + 1)
+
+		self:setState({
+			cellSize = cellWidth,
+			canvasHeight = canvasHeight,
+		})
+	end
+end
+
+function ShopPage:render()
+	local props = self.props
+	local state = self.state
+	local padding = props.padding or 0
+	local cellSize = state.cellSize
+	local canvasHeight = state.canvasHeight
+
+	local frameProps = {}
+
+	for key, value in pairs(props) do
+		if not excludePropsForFrame[key] then
+			frameProps[key] = value
+		end
+	end
```

Comment:
I think this is generally an anti-pattern. I like to explicitly scope all the props intended for the container to a sub-table, which is less error-prone!